modules/alloy/src/main/ts/ephox/alloy/behaviour/blocking/BlockingState.ts

```diff
@@ -0,0 +1,42 @@
+import { Cell, Optional } from '@ephox/katamari';
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import { BehaviourState, nuState } from '../common/BehaviourState';
+
+export interface BlockingState extends BehaviourState {
+  setBlocker: (actual: AlloyComponent, destroy: () => void) => void;
+  clearBlocker: () => void;
+  getBlocked: () => boolean;
+  setBlocked: (blocked: boolean) => void;
+}
+
+const init = (): BlockingState => {
+  const blocker = Cell(Optional.none<{ actual: AlloyComponent; destroy: () => void }>());
+  const blocked = Cell(false);
+
+  const readState = () => ({
```

Comment:
`readState` needs to return a string representation or something that can be JSON serialized, as it's used in the devtools view.

- [ ] Return a string based representation

Reply:
Done